Ardupilog.m

```diff
@@ -146,12 +150,28 @@ function delete(obj)
                         error('Unexpected comparison result');
                     end
                 end
-
-                msgLen = obj.fmt_cell{i,3};
-                data = obj.isolateMsgData(msgId,msgLen,allHeaderCandidates);
+                
+                % If message not filtered, store it
                 obj.(msgName).storeData(data');
             end
             
+            % Construct the LineNo for the whole log
+            LineNo_ndx_vec = sort(vertcat(obj.valid_msgheader_cell{:,2}));
+            LineNo_vec = [1:length(LineNo_ndx_vec)]';
+            % For each LogMsgGroup which wasn't filtered
+            for i = 1:size(obj.valid_msgheader_cell,1)
+                % Find msgName from msgId in 1st column
+                msgId = obj.valid_msgheader_cell{i,1};
+                row_in_fmt_cell = vertcat(obj.fmt_cell{:,1})==msgId;
+                msgName = obj.fmt_cell{row_in_fmt_cell,2};
+
+                % Pick out the correct line numbers
+                msg_LineNo = LineNo_vec(ismember(LineNo_ndx_vec, obj.valid_msgheader_cell{i,2}));
+                
+                % Write to the LogMsgGroup
+                obj.(msgName).setLineNo(msg_LineNo);
+            end
+
```

Comment:
So what you do here is: (correct me if I'm wrong)
1. You sort the locations of all messages
2. For each filtered message you search the line numbers of the corresponding messages and store them in the LineNo property